services/frontendhost/frontend_test.go

```diff
@@ -663,6 +663,72 @@ func (s *FrontendHostSuite) TestFrontendHostCreateConsumerGroup() {
 	}
 }
 
+// TestFrontendHostCreateConsumerGroupStartFrom tests whether CreateConsumerGroup is able to
+// detect and appropriately convert various startFrom units.
+func (s *FrontendHostSuite) TestFrontendHostCreateConsumerGroupStartFrom() {
+
+	testPath := uuid.New()
+	testCG := s.generateKey("/bar/CGName")
+	frontendHost, ctx := s.utilGetContextAndFrontend()
+
+	// set start-from to, say, four weeks ago
+	startFromTime := time.Now().Add(-4 * 7 * 24 * time.Hour)
+
+	startFromSeconds := startFromTime.Unix()
+	startFromMillis := startFromSeconds * int64(1000)
+	startFromMicros := startFromSeconds * int64(1000000)
+	startFromNanos := startFromSeconds * int64(1000000000)
+
+	testStartFrom := func(startFrom int64) {
+		req := c.NewCreateConsumerGroupRequest()
+		req.DestinationPath = common.StringPtr(testPath)
+		req.ConsumerGroupName = common.StringPtr(testCG)
+
+		if startFrom == 0 {
+			req.StartFrom = common.Int64Ptr(0)
+		} else {
+			req.StartFrom = common.Int64Ptr(startFromNanos)
+		}
+
+		cgDesc := cgCreateRequestToDesc(req)
+		frontendHost.writeCacheDestinationPathForUUID(destinationUUID(cgDesc.GetDestinationUUID()), testPath)
+
+		dlqPath, _ := common.GetDLQPathNameFromCGName(testCG)
+		destDesc := &shared.DestinationDescription{
+			Path:            common.StringPtr(dlqPath),
+			DestinationUUID: common.StringPtr(uuid.New()),
+		}
+
+		// create destination is needed because of dlq destination been created at time of consumer group creation
+		s.mockController.On("CreateDestination", mock.Anything, mock.Anything).Return(destDesc, nil).Run(func(args mock.Arguments) {
```

Comment:
see my response above; i still think it is better to test at the CreateConsumerGroup API level.